### PR TITLE
refactor interceptor resolver: use mutex in event struct

### PR DESCRIPTION
**Pull request by ssd04**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- Refactor debug interceptor resolver in order to avoid using mutex for all cacher interactions
  
## Proposed Changes
- remove mutex for cache
- use mutex for internal event structure

## Testing procedure
- Standard system test

### CODE REVIEW

File: debug/resolver/interceptorResolver.go [hunk 1]
```diff
@@ -42,8 +43,19 @@ type event struct {
...
+func (ev *event) NumPrints() int {
+	ev.mutEvent.Lock()
+	defer ev.mutEvent.Unlock()
+
+	return ev.numPrints
+}
+
 // Size returns the number of bytes taken by an event line
 func (ev *event) Size() int {
+	ev.mutEvent.Lock()
```
Comment (AdoAdoAdo) [inline debug/resolver/interceptorResolver.go:56]: can be RLock

File: debug/resolver/interceptorResolver.go [hunk 2]
```diff
@@ -53,6 +65,9 @@ func (ev *event) Size() int {
 }
 
 func (ev *event) String() string {
+	ev.mutEvent.Lock()
```
Comment (AdoAdoAdo) [inline debug/resolver/interceptorResolver.go:68]: can be RLock

File: debug/resolver/interceptorResolver.go [hunk 3]
```diff
@@ -42,8 +43,19 @@ type event struct {
 	timestamp    int64
 }
 
+// NumPrints returns the current number of prints
+func (ev *event) NumPrints() int {
+	ev.mutEvent.Lock()
```
Comment (AdoAdoAdo) [inline debug/resolver/interceptorResolver.go:48]: can be RLock

File: debug/resolver/interceptorResolver.go [hunk 4]
```diff
@@ -185,15 +196,19 @@ func (ir *interceptorResolver) incrementNumOfPrints() {
 			continue
 		}
 
+		ev.mutEvent.Lock()
 		ev.numPrints++
-		ir.cache.Put(key, ev, ev.Size())
+		ev.mutEvent.Unlock()
```
Comment (AdoAdoAdo) [inline debug/resolver/interceptorResolver.go:201]: :+1:

File: debug/resolver/interceptorResolver.go [hunk 5]
```diff
@@ -185,15 +196,19 @@ func (ir *interceptorResolver) incrementNumOfPrints() {
...
 	}
 }
 
 //TODO replace this with a call to Query(search) when a suitable conditional parser will be used. Also replace config parameters
 // with a query string so it will be more extensible
 func (ir *interceptorResolver) getStringEvents(maxNumPrints int) []string {
 	acceptEvent := func(ev *event) bool {
+		ev.mutEvent.Lock()
```
Comment (AdoAdoAdo) [inline debug/resolver/interceptorResolver.go:209]: RLock

File: debug/resolver/interceptorResolver.go [hunk 6]
```diff
@@ -239,10 +251,11 @@ func (ir *interceptorResolver) logRequestedData(topic string, hash []byte, numRe
 		return
 	}
 
+	req.mutEvent.Lock()
 	req.numReqCross += numReqCross
 	req.numReqIntra += numReqIntra
 	req.timestamp = ir.timestampHandler()
-	ir.cache.Put(identifier, req, req.Size())
+	req.mutEvent.Unlock()
```
Comment (AdoAdoAdo) [inline debug/resolver/interceptorResolver.go:258]: :+1:

File: debug/resolver/interceptorResolver.go [hunk 7]
```diff
@@ -318,6 +327,8 @@ func (ir *interceptorResolver) Query(search string) []string {
 	acceptEvent := func(ev *event) bool {
 		//TODO replace this rudimentary search pattern with something like
 		// github.com/oleksandr/conditions
+		ev.mutEvent.Lock()
```
Comment (AdoAdoAdo) [inline debug/resolver/interceptorResolver.go:330]: RLock

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

File: debug/resolver/interceptorResolver.go (hunk 1 above)
Comment (ssd04) [reply debug/resolver/interceptorResolver.go]: thanks, done

File: debug/resolver/interceptorResolver.go (hunk 2 above)
Comment (ssd04) [reply debug/resolver/interceptorResolver.go]: done

File: debug/resolver/interceptorResolver.go (hunk 3 above)
Comment (ssd04) [reply debug/resolver/interceptorResolver.go]: done

**Review by AdoAdoAdo — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**